components/esp_event/esp_event.c

```diff
@@ -526,30 +526,30 @@ esp_err_t esp_event_loop_run(esp_event_loop_handle_t event_loop, TickType_t tick
 
         bool exec = false;
 
-        esp_event_handler_instance_t *handler;
+        esp_event_handler_instance_t *handler, *temp_handler;
         esp_event_loop_node_t *loop_node;
-        esp_event_base_node_t *base_node;
-        esp_event_id_node_t *id_node;
+        esp_event_base_node_t *base_node, *temp_base;
+        esp_event_id_node_t *id_node, *temp_id_node;
 
         SLIST_FOREACH(loop_node, &(loop->loop_nodes), next) {
```

Comment:
This seems to also need the safe foreach variant.

Reply:
Sorry for the long silence, but you're right. I've also seen your fixup.